x-pack/plugins/index_management/public/application/sections/home/data_stream_list/data_stream_table/data_stream_table.tsx

```diff
@@ -151,7 +153,14 @@ export const DataStreamTable: React.FunctionComponent<Props> = ({
     ),
     truncateText: true,
     sortable: true,
-    render: (lifecycle: DataStream['lifecycle']) => getLifecycleValue(lifecycle, INFINITE_AS_ICON),
+    render: (lifecycle: DataStream['lifecycle'], dataStream) => (
+      <ConditionalWrap
+        condition={isDataStreamFullyManagedByILM(dataStream)}
```

Comment:
Do you think there is any concern in terms of performance? If I understand correctly, we are checking every index of every data stream here.

Reply:
Good call! I've wrapped the computed property around a useMemo now